packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -143,16 +146,16 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
   }
 
   Future<GraphQLResponse<T>> _getMethodChannelResponse<T>({
-    @required methodName,
-    @required GraphQLRequest<T> request,
+    required methodName,
+    required GraphQLRequest<T> request,
   }) async {
     try {
-      final Map<String, dynamic> result =
-          await _channel.invokeMapMethod<String, dynamic>(
+      final Map<String, dynamic>? result =
+          (await (_channel.invokeMapMethod<String, dynamic>(
         methodName,
         request.serializeAsMap(),
-      );
-
+      )));
+      if (result == null) throw nullMapFromMethodChannelException;
```

Comment:
noob question: Why not just `throw AmplifyException(AmplifyExceptionMessages.nullReturnedFromMethodChannel);` here? Is it an optimization, convention, or is there some value to throwing same literal error every time?

Reply:
We can do that too.  

I figured it would be cleaner in the code to have a dedicated type for that instead of repeating the code throughout.

Reply:
no change needed, just curious